### PR TITLE
Add default payment method to PaymentSheet loading analytic

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
Sends `selected_lpm` in the `mc_load_succeeded` analytic. This was pretty difficult, since the code that determines the default is inside the SavedPaymentOptionsVC which isn't necessarily loaded until after `mc_load_succeeded` is sent.

To make this work I extracted out the code to determine the SavedPaymentOptionsVC view models and default selection, and call it in load to know the value ahead of time. 

The long term fix might be to continue to create view models in the load path and then *flow that through* to the SavedPaymentOptionsVC - and more generally, generate view models for *all* the UI at load and then flow that through to the views.  

Also renames BI's equivalent analytic's param from `default_payment_method` to `selected_lpm`.

## Motivation
https://jira.corp.stripe.com/browse/MOBILESDK-1562

## Testing
Unit tests + manual testing for "apple_pay", "link", "card", "us_bank_account", "none"

## Changelog
Not user facing

### CODE REVIEW

File: Stripe/StripeiOS/Source/STPAnalyticsClient+BasicUI.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ extension STPPaymentContext {
             }()
             let params: [String: Any] = [
                 "duration": duration,
-                "default_payment_method": defaultPaymentMethod,
+                "selected_lpm": defaultPaymentMethod,
```
Comment (porter-stripe) [inline Stripe/StripeiOS/Source/STPAnalyticsClient+BasicUI.swift:54]: Will this break existing queries that expect default_payment_method to be populated?

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetLoader.swift [hunk 2]
```diff
@@ -72,14 +74,30 @@ final class PaymentSheetLoader {
...
                 let isLinkEnabled = isLinkEnabled(intent: intent, configuration: configuration)
-                analyticsClient.logPaymentSheetEvent(event: .paymentSheetLoadSucceeded,
-                                                                     duration: Date().timeIntervalSince(loadingStartDate))
+                let isApplePayEnabled = StripeAPI.deviceSupportsApplePay()
+                    && configuration.applePay != nil
+                    && intent.isApplePayEnabled
+
+                // Send load finished analytic
+                // This is hacky; the logic to determine the default selected payment method belongs to the SavedPaymentOptionsViewController. We invoke it here just to report it to analytics before that VC loads.
+                let (defaultSelectedIndex, paymentOptionsViewModels) = SavedPaymentOptionsViewController.makeViewModels(
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetLoader.swift:86]: Does the default pm need to be sent in the loaded event? Seems like we have access to an initialized `SavedPaymentOptionsViewController` in `PaymentSheetViewController` and `PaymentSheetFlowControllerViewController`, which is where we also log the show event. Can we attach the default to the show event? Would that be less hacky?

File: Stripe/StripeiOS/Source/STPAnalyticsClient+BasicUI.swift (hunk 1 above)
Comment (yuki-stripe) [reply Stripe/StripeiOS/Source/STPAnalyticsClient+BasicUI.swift:54]: No, `default_payment_method` was added earlier this week.

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetLoader.swift (hunk 2 above)
Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetLoader.swift:86]: I think so. For FlowController it's possible (likely, even, if you have a default PM) to never show the sheet before completing payment.

**Review by porter-stripe — APPROVED** [review]

**Review by porter-stripe — DISMISSED** [review]

**Review by porter-stripe — APPROVED** [review]

**Pull request merged**